opal/mca/accelerator/cuda/accelerator_cuda.c

```diff
@@ -578,6 +579,45 @@ static int accelerator_cuda_get_device(int *dev_id)
     return 0;
 }
 
+static int accelerator_cuda_get_device_pci_attr(int dev_id, opal_accelerator_pci_attr_t *pci_attr)
+{
+    CUresult result;
+    int ret;
+    static const int PCI_BUS_ID_LENGTH = 13;
+    char pci_bus_id[PCI_BUS_ID_LENGTH];
+    char domain_id[5] = {0}, bus_id[3] = {0}, device_id[3] = {0}, function_id[2] = {0};
+
+    if (NULL == pci_attr) {
+        return OPAL_ERR_BAD_PARAM;
+    }
+
+    result = cuDeviceGetPCIBusId(pci_bus_id, PCI_BUS_ID_LENGTH, dev_id);
+
+    if (CUDA_SUCCESS != result) {
+        opal_output_verbose(5, opal_accelerator_base_framework.framework_output,
+                            "CUDA: Failed to get device PCI bus id");
+        return OPAL_ERROR;
+    }
+
+    ret = sscanf(pci_bus_id, "%4s:%2s:%2s.%s", domain_id, bus_id, device_id, function_id);
```

Comment:
The last string (function_id) in sscanf doesn't have a length. Is it variable size?

Reply:
Thank you. Fixed.